### PR TITLE
avoid call to HTMLMediaElement#load when peaks are given

### PR DESCRIPTION
**Pull request by osheroff**
calling `.load()` explicitly on a mediaelement entirely defeats the "preload" setting and makes it impossible to avoid downloading data before playback, which is often what the user is trying to do by providing peak data. 

When HTMLMediaElement#load is called, all of the major browsers make some request for the file. 
Chrome and FF make a range request for the entire file, but only report downloading them partially.  Safari reports downloading the file in its entirety.

### CODE REVIEW

**Comment by thijstriemstra** [comment]
thanks @osheroff, could you check if there are any tickets that get fixed by this change (and add them to the PR description)?

**Comment by osheroff** [comment]
I would have guessed there would have been more.

- maybe affects: https://github.com/katspaugh/wavesurfer.js/issues/740
- will help: https://github.com/katspaugh/wavesurfer.js/issues/1918
- partially: https://github.com/katspaugh/wavesurfer.js/issues/1866 (here there are two issues, the first is a problem with the webaudio backend but he reports being unable to use the mediaelement backend due to the issue I've been seeing)

There's also a whole host of people talking about memory issues etc, but they're likely using the webaudio backend.

**Comment by thijstriemstra** [comment]
can you add a changelog entry?

**Comment by thijstriemstra** [comment]
ps. it seems only Chrome dies during the CI builds, Firefox seems fine. We can disable Chrome on CI for a while and try again when it's back up and stable. Not being able to trust the CI results is annoying.

**Comment by marizuccara** [comment]
Maybe it can be done not when `!peaks` but when `preload `attribute is different from `none` ?

**Comment by marizuccara** [comment]
Hi maybe the check can be done not if `!peaks` but for example if `preload` attribute is of type `none`?

**Comment by osheroff** [comment]
> Hi maybe the check can be done not if !peaks but for example if preload attribute is of type none?

@marizuccara I think it's a good point, but maybe it should be both `!peaks && (preload == 'none')`?  If we're not given server-generated peaks, I think we always have to call `.load` in order to trigger the waveform drawing (at least that's what the comment in the file seemed to indicated).

**Comment by osheroff** [comment]
any further comments here?  @marizuccara I modified it to check if `preload == 'none'`

File: CHANGES.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ wavesurfer.js changelog
 4.0.0 (21.06.2020)
 ------------------
 
+- Don't call HTMLMediaElement#load when given peaks and preload == 'none'.
+  Prevents browsers from pre-fetching audio (#1969)
```
Comment (thijstriemstra) [inline CHANGES.md:8]: 4.0.0 is already released, you can add a new entry, e.g. 4.1.0.

Comment (osheroff) [reply CHANGES.md:8]: k fixed.

**Comment by thijstriemstra** [comment]
branch has merge conflict.

**Comment by osheroff** [comment]
Ok I fixed the merge conflict.  The system of putting changelog entries into PRs is a bit problematic, as every time another PR is merged into master my branch goes back into conflict (this is like the 3rd time I've had to update CHANGES.md on this PR).  I'm wondering if there's a better solution...

**Comment by thijstriemstra** [comment]
>  The system of putting changelog entries into PRs is a bit problematic, as every time another PR is merged into master my branch goes back into conflict

We can use a section like this at top to reduce merge conflicts:

```
Next (unreleased)
------------------
```

File: src/mediaelement.js [hunk 2]
```diff
@@ -160,7 +160,7 @@ export default class MediaElement extends WebAudio {
         elt.controls = this.params.mediaControls;
         elt.autoplay = this.params.autoplay || false;
 
-        this._load(elt, peaks);
+        this._load(elt, peaks, elt.preload);
```
Comment (thijstriemstra) [inline src/mediaelement.js:163]: What if `elt.preload` is undefined?

Comment (osheroff) [reply src/mediaelement.js:163]: good question. 

It won't be undefined.  If you don't set the HTML tag, in chrome/ff, the `preload` element will default to `""`.  In Safari, it defaults to `"auto"`.  The behavior is also similar: chrome/ff don't preload at all, and Safari first does a range request and then ends up preloading a whole 2mb file. 

I can detect these cases here and change our `.load` behavior accordingly, but I'm really not sure what the user's intention really is.  My guess is that if they're providing peak data they almost *never* want a preload, but that's just a guess...

**Pull request merged**